src/opencl_common.c

```diff
@@ -1867,6 +1867,8 @@ void opencl_find_best_lws(size_t group_size_limit, int sequential_id,
 			fprintf(stderr, " %ss%s\n", ns2string(sumRunTime),
 			    ((double)(sumRunTime) / kernelExecTimeNs < 0.997)
 			        ? "+" : "");
+		if (sumRunTime > 2 * kernelExecTimeNs)
+			break;
```

Comment:
This is for the issue [reported here](https://github.com/openwall/john/issues/4684#issuecomment-824831938), which was intermittent. I figured if we *ever* get over twice as long duration from doubling LWS but with same GWS, we can't be on the way to anything good, right?

Reply:
Doubled duration with same GWS means twice lower speeds, so, yes, I guess it means the doubled LWS wasn't a move in the right direction. I wonder, though, whether exceptions exist with LWS/GWS values way out of the device's optimal range, where further GWS adjustment would bring them into the optimal range. Hopefully not.

Maybe this should be a separate commit with its own description.